### PR TITLE
feat: global messages context provider

### PR DESCRIPTION
**Pull request by mortennordseth**
This includes a new context provider for global messages. The context provider is not in use yet, but will be when completing https://github.com/AtB-AS/kundevendt/issues/17725.

### CODE REVIEW

**Review by adriansberg — APPROVED** [review]

File: src/modules/global-messages/index.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+}
+
+const globalMessageConverter = {
+  toFirestore(_any: any) {
+    throw new Error('Not implemented or used');
+  },
+  fromFirestore(
+    snapshot: QueryDocumentSnapshot<GlobalMessageType>,
+  ): GlobalMessageType {
+    return snapshot.data() as GlobalMessageType;
```
Comment (mikaelbr) [inline src/modules/global-messages/index.tsx:93]: This should be fixed to be safely parsed at somepoint™

Comment (mortennordseth) [reply src/modules/global-messages/index.tsx]: The future is now ...

File: src/modules/global-messages/converters.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      body,
+      title,
+      isDismissable,
+      startDate,
+      endDate,
+    };
+  },
+};
+
+function isGlobalMessageType(data: any): data is GlobalMessageType {
```
Comment (mortennordseth) [inline src/modules/global-messages/converters.ts:89]: Should probably redo this function to instead return a boolean on whether data contains the necessary fields to be a global message type

**Review by adriansberg — APPROVED** [review]
LGTM

File: src/modules/global-messages/converters.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    const body = mapToLocalizedStringArray(data.body);
+    const title = mapToLocalizedStringArray(data.title);
+    const context = mapToContexts(data.context);
+    const type = mapToMessageType(data.type);
+    const subtle = data.subtle ?? false;
+    const isDismissable = data.isDismissable ?? false;
+    const startDate = data.startDate;
+    const endDate = data.endDate;
+
+    if (!body) return;
```
Comment (adriansberg) [inline src/modules/global-messages/converters.ts:70]: Could combine these to one statement.

File: src/modules/global-messages/index.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+    querySnapshot.forEach((doc) => {
+      const data = doc.data();
+      if (data) {
+        activeGlobalMessages.push(data);
+      }
+    });
+
+    updateActiveGlobalMessages(activeGlobalMessages);
+    console.log(
```
Comment (adriansberg) [inline src/modules/global-messages/index.tsx:83]: Should remove this at some point.

**Review by adriansberg — APPROVED** [review]

File: src/modules/global-messages/converters.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+    const active = data.active;
+    const body = mapToLocalizedStringArray(data.body);
+    const title = mapToLocalizedStringArray(data.title);
+    const context = mapToContexts(data.context);
+    const type = mapToMessageType(data.type);
+    const subtle = data.subtle ?? false;
+    const isDismissable = data.isDismissable ?? false;
+    const startDate = data.startDate;
+    const endDate = data.endDate;
```
Comment (mikaelbr) [inline src/modules/global-messages/converters.ts:70]: Feel like ideally this should be refactored to zod parsers – but not needed right now.

**Pull request merged**